Rpackage/R/fitAndScore.R

```diff
@@ -0,0 +1,191 @@
+                                       maxdepth = as.numeric(parameters$maxdepth), margin = as.numeric(parameters$margin), 
+                                       loss = loss, min_sample = as.numeric(parameters$min_sample))
+  }
+  
+  ### master tree
+  master_tree <- mmit(target.mat, feature.mat,  
+                      maxdepth = as.numeric(parameters$maxdepth), margin = as.numeric(parameters$margin), 
+                      loss = loss, min_sample = as.numeric(parameters$min_sample))
+  
+  ### if pruning
+  if(pruning){
+    # Get the pruned master and cross-validation trees
+    master_data <- mmit.pruning(master_tree)
+    master_alphas <- rev(unlist(lapply(master_data, function(x) x$alpha)))
+    master_pruned_trees <- rev(lapply(master_data, function(x) x$tree))
+    
+    fold_alphas <- NULL
+    fold_prune_trees <- NULL
+    for(i in 1 : n_folds){
+      fold_data <- mmit.pruning(fold_tree[[i]])
+      fold_alphas[[i]] <- rev(unlist(lapply(fold_data, function(x) x$alpha)))
+      fold_prune_trees[[i]] <- rev(lapply(fold_data, function(x) x$tree))
+    }
+    
+    ### alphas list should not contain repeating alpha
+    master_alphas <- master_alphas[!duplicated(master_alphas)]
+    fold_alphas <- fold_alphas[!duplicated(fold_alphas)]
+    
+    
+    # Compute the test risk for all pruned trees of each fold
+    alpha_path_score <- NULL
+    for(i in 1 : n_folds){
+      for(j in 1 : length(fold_prune_trees[[i]])){
+        ### convert pruned tree list to partynode
+        node <- fold_prune_trees[[i]][[j]]
+        
+        ### predictions of the tree for test data
+        prediction <- mmit.predict(node, feature.mat[fold_split_idx$test[i,],])
+        
+
+        ###error calc
+        fold_test_scores <- scorer(target.mat[fold_split_idx$test[i,],], prediction)
+        
+        ### creating a dataframe with init alpha, final alpha, score value
+        if(j < (length(fold_alphas[[i]]))){
+          if(length(alpha_path_score)<i){
+            alpha_path_score[[i]] <- c(fold_alphas[[i]][j], fold_alphas[[i]][j + 1], fold_test_scores)
+          }
+          else{
+            alpha_path_score[[i]] <- rbind(alpha_path_score[[i]], c(fold_alphas[[i]][j], fold_alphas[[i]][j + 1], fold_test_scores))
+            }
+          }
+        else{
+          
+          if(length(alpha_path_score)<i){
+            alpha_path_score[[i]] <- c(fold_alphas[[i]][j], Inf, fold_test_scores)
+          }
+          else{
+            alpha_path_score[[i]] <- rbind(alpha_path_score[[i]], c(fold_alphas[[i]][j], Inf, fold_test_scores))
+          }
+        }
+      }
+      colnames(alpha_path_score[[i]]) <- c("init alpha", " final alpha", "score")
+      alpha_path_score[[i]] <- as.data.frame(alpha_path_score[[i]])
+    }
+    
+    # Prune the master tree based on the CV estimates
+    alphas <- NULL
+    alpha_cv_scores <- NULL
+    alpha_train_scores <- NULL
+    alpha_train_objective_values <- NULL
+    best_alpha <- -1
+    best_score <- Inf
```

Comment:
Use scorer "worst" attribute to initialize.

Reply:
done